File: Go/Array/8958.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	var tc int
	reader := bufio.NewReader(os.Stdin)
	fmt.Fscan(reader, &tc)

	for i := 0; i < tc; i++ {
		prev_score := 0
		total := 0
		var input string
		fmt.Fscan(reader, &input)
		a := strings.Split(input, "")
		for _, val := range a {
			if val == "O" {
				prev_score++
				total += prev_score
			} else {
				prev_score = 0
			}
		}
		fmt.Println(total)
	}

}

File: python/Array/8958.py
import sys

if __name__ == "__main__":
    tc=int(input())
    for i in range(tc):
        _str=input()
        prev_score=0
        total=0
        for val in _str:
            if(val=='O'):
                prev_score+=1
                total+=prev_score
            else:
                prev_score=0
        print(total)
